bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/EcobeeGenericBindingProvider.java

```diff
@@ -0,0 +1,313 @@
+ * <p>
+ * Changes the backlight sleep intensity on all thermostats at the lake house (meaning, all thermostats registered to
+ * the lakehouse Ecobee account).</li>
+ * </ul>
+ * 
+ * @author John Cocula
+ * @since 1.7.0
+ */
+public class EcobeeGenericBindingProvider extends AbstractGenericBindingProvider implements EcobeeBindingProvider {
+
+	private static class EcobeeBindingConfig implements BindingConfig {
+		String userid;
+		String thermostatIdentifier;
+		String property;
+		boolean inBound = false;
+		boolean outBound = false;
+
+		public EcobeeBindingConfig(final String userid, final String thermostatIdentifier, final String property,
+				final boolean inBound, final boolean outBound) {
+			this.userid = userid;
+			this.thermostatIdentifier = thermostatIdentifier;
+			this.property = property;
+			this.inBound = inBound;
+			this.outBound = outBound;
+		}
+
+		@Override
+		public String toString() {
+			return "EcobeeBindingConfig [userid=" + this.userid + "thermostatIdentifier=" + this.thermostatIdentifier
+					+ ", property=" + this.property + ", inBound=" + this.inBound + ", outBound=" + this.outBound + "]";
+		}
+	}
+
+	private static Logger logger = LoggerFactory.getLogger(EcobeeGenericBindingProvider.class);
+
+	private static final Pattern CONFIG_PATTERN = Pattern.compile(".\\[(.*)#(.*)\\]");
+
+	// the first character in the above pattern
+	private static final String IN_BOUND = "<";
+	private static final String OUT_BOUND = ">";
+	private static final String BIDIRECTIONAL = "=";
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "ecobee";
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getUserid(final String itemName) {
+		final EcobeeBindingConfig config = (EcobeeBindingConfig) this.bindingConfigs.get(itemName);
+		return config != null ? config.userid : null;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getThermostatIdentifier(final String itemName) {
+		EcobeeBindingConfig config = (EcobeeBindingConfig) this.bindingConfigs.get(itemName);
+		return config != null ? config.thermostatIdentifier : null;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getProperty(final String itemName) {
+		EcobeeBindingConfig config = (EcobeeBindingConfig) this.bindingConfigs.get(itemName);
+		return config != null ? config.property : null;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean isInBound(final String itemName) {
+		EcobeeBindingConfig config = (EcobeeBindingConfig) this.bindingConfigs.get(itemName);
+		return config != null ? config.inBound : false;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean isOutBound(final String itemName) {
+		EcobeeBindingConfig config = (EcobeeBindingConfig) this.bindingConfigs.get(itemName);
+		return config != null ? config.outBound : false;
+	}
+
+	// Injected by the OSGi Container through the setItemRegistry and
+	// unsetItemRegistry methods.
+	private ItemRegistry itemRegistry;
```

Comment:
please move the ItemRegistry member to the Binding since you are not using the Registry here